delphi/app/pki_model.py

```diff
@@ -0,0 +1,106 @@
+from delphi.app.graph import run_optimization
+
+
+class PKI:
+    """
+    PKI is a class to represent the building PKI.
+
+    Attributes:
+        vertices: a dictionary of nodes and their enabled status
+        l2vertices: a dictionary of nodes and their enabled status
+        l3vertices: a dictionary of nodes and their enabled status
+        s1: the people count of room s1
+        s2: the people count of room s2
+        s3: the people count of room s3
+        desired_flow: the amount of people that need to exit pki
+        edges: a dictionary of edges and their weights
+        l2edges: a dictionary of edges and their weights
+        l3edges: a dictionary of edges and their weights
+    """
+
+    def __init__(self):
+        """
+        PKI constructor
+        """
+        self.vertices = {'t1': True, 't2': True, 's1': True, 's2': True, 's3': True, 'h1': True, 'h2': True, 'h5': True}
+        self.l2vertices = {'t3': True, 't4': True, 'h3': True, 'h4': True}
+        self.l3vertices = {'h6': True}
+        self.s1 = 25
+        self.s2 = 30
+        self.s3 = 20
```

Comment:
create a room count dict that stores this information? or better yet, have vertices store their room count and status.

Reply:
Having vertices store their room count and status would be best

Reply:
s1, s2, and s3 only serve a single purpose in this model. I don't think we need to go as far as to have vertices keep track of their count. It would contradict the graph model and be less efficient. I can go ahead and make a dictionary for s1, s2, and s3 if that is okay with you Piyush.

Reply:
We need to way to keep track of the count of people in each room though, since that info is also sent up to oracle and the flow network analyzer will need that information as it gets information about new rooms

Reply:
This info is being kept track of through edges. That is how the network flow problem works. The model already accounts for this. These variables are just being used to set the proper edge weights.

Reply:
The data is incrementally given though, so if the counts need to be maintained until the next time that particular room's count is updated. For example, this will get the counts for room A, and then calculate the flow network, and then later will get the counts for room B, and need to calculate the flow using both the previous data received about A and the new data received about B. This necessitates needing to store that information somehow.

Reply:
Yes. I do not see the problem with the way it is now. the update_room function handles updating the correct edges and maintaining old room counts that havent been changed.